src/pymorize/global_attributes.py

```diff
@@ -0,0 +1,312 @@
+file.
+
+Use the `get_global_attributes` method on `DataRequestVariable` to get the global attributes.
+"""
+
+import re
+import uuid
+
+import xarray as xr
+
+from .factory import MetaFactory, create_factory
+
+_parent_fields = (
+    "branch_method",
+    "branch_time_in_child",
+    "branch_time_in_parent",
+    "parent_experiment_id",
+    "parent_mip_era",
+    "parent_source_id",
+    "parent_time_units",
+    "parent_variant_label",
+)
+
+
+class GlobalAttributes(metaclass=MetaFactory):
+    @classmethod
+    def get_global_attributes(cls, attrs_map_on_rule: dict) -> dict:
+        """Get the global attributes for a dataset."""
+        raise NotImplementedError
+
+
+class CMIP6GlobalAttributes(GlobalAttributes):
+    """GlobalAttributes for CMIP6"""
+
+    def __init__(self, cv):
+        """
+        Parameters
+        ----------
+        cv : ControlledVocabularies
+            Controlled Vocabularies for CMIP6.
+        """
+        self.cv = cv
+
+    def _parse_variant_label(self, label: str) -> dict:
+        """
+        Extracts indices values from variant label.
+        `label` must be of the form "r<int>i<int>p<int>f<int>".
+        Example
+        -------
+        >>> label = "r1i1p1f1"
+        >>> _parse_variant_label(label)
+        {'realization_index': 1, 'initialization_index': 1, 'physics_index': 1, 'forcing_index': 1,}
+        """
+        pattern = re.compile(
+            r"r(?P<realization_index>\d+)"
+            r"i(?P<initialization_index>\d+)"
+            r"p(?P<physics_index>\d+)"
+            r"f(?P<forcing_index>\d+)"
+            r"$"
+        )
+        if label is None:
+            raise ValueError(
+                f"`label` must be of the form 'r<int>i<int>p<int>f<int>', Got: {label}"
+            )
+        d = pattern.match(label)
+        if d is None:
+            raise ValueError(
+                f"`label` must be of the form 'r<int>i<int>p<int>f<int>', Got: {label}"
+            )
+        d = {name: int(val) for name, val in d.groupdict().items()}
+        return d
+
+    def _source_id_related(self, attrs_map_on_rule: dict) -> dict:
+        source_id = attrs_map_on_rule.get("source_id")
+        cv = self.cv["source_id"][source_id]
+        cv_institution_ids = cv["institution_id"]
+        user_institution_id = attrs_map_on_rule.get("institution_id", None)
+        if user_institution_id:
+            if user_institution_id not in cv_institution_ids:
+                raise ValueError(
+                    f"Institution ID '{user_institution_id}' is not valid. "
+                    f"Allowed values: {cv_institution_ids}"
+                )
+            selected_institution_id = user_institution_id
+        else:
+            # No user-provided institution_id; infer it
+            if len(cv_institution_ids) > 1:
+                raise ValueError(
+                    f"Multiple institution IDs found: {cv_institution_ids}. "
+                    "Please specify one in the rule."
```

Comment:
```suggestion
                    "Please specify a ``institution_id`` in the rule."
```
So that they know which particular key they should be using.